File: Server/ConferenceRoom.Domain/Entities/Booking.cs
public sealed class Booking : BaseEntity
{
    public Booking(
        Guid roomId,
        Guid userId,
        DateTime startTimeUtc,
        DateTime endTimeUtc,
        string purpose)
    {
        RoomId = roomId;
        UserId = userId;
        StartTimeUtc = startTimeUtc;
        EndTimeUtc = endTimeUtc;
        Purpose = purpose;
        Status = BookingStatus.Active;

        RaiseDomainEvent(new BookingCreatedDomainEvent(
            Id,
            RoomId,
            UserId,
            StartTimeUtc,
            EndTimeUtc,
            Purpose,
            DateTime.UtcNow));
    }

    public Guid RoomId { get; private set; }
    public Room Room { get; private set; } = null!;

    public Guid UserId { get; private set; }
    public User User { get; private set; } = null!;

    public DateTime StartTimeUtc { get; private set; }
    public DateTime EndTimeUtc { get; private set; }

    public string Purpose { get; private set; } = string.Empty;

    public BookingStatus Status { get; private set; }

    public bool IsActive => Status == BookingStatus.Active;

    public void Update(
        Guid roomId,
        DateTime startTimeUtc,
        DateTime endTimeUtc,
        string purpose)
    {
        RoomId = roomId;
        StartTimeUtc = startTimeUtc;
        EndTimeUtc = endTimeUtc;
        Purpose = purpose;

        RaiseDomainEvent(new BookingUpdatedDomainEvent(
            Id,
            RoomId,
            UserId,
            StartTimeUtc,
            EndTimeUtc,
            Purpose,
            DateTime.UtcNow));
    }

    public void Cancel()
    {
        Status = BookingStatus.Cancelled;

        RaiseDomainEvent(new BookingCancelledDomainEvent(
            Id,
            RoomId,
            UserId,
            StartTimeUtc,
            EndTimeUtc,
            DateTime.UtcNow));
    }
}

File: Client/src/features/bookings/types/booking.ts
export type BookingStatus = 'Active' | 'Cancelled';

export type RecurrenceType = 'Daily' | 'Weekly' | 'Monthly';

export type Booking = {
  id: string;
  bookingId?: string;
  roomId: string;
  roomName?: string;
  roomCapacity?: number;
  userId: string;
  userName?: string;
  startTimeUtc: string;
  endTimeUtc: string;
  purpose: string;
  status: BookingStatus;
  recurringBookingSeriesId?: string | null;
};

export type CreateBookingRequest = {
  roomId: string;
  startTimeUtc: string;
  endTimeUtc: string;
  purpose: string;
};

export type UpdateBookingRequest = CreateBookingRequest;

export type QuickBookingRequest = {
  startTimeUtc: string;
  endTimeUtc: string;
  numberOfPeople: number;
  purpose: string;
};

export type CreateRecurringBookingRequest = {
  roomId: string;
  startTimeUtc: string;
  endTimeUtc: string;
  purpose: string;
  recurrenceType: RecurrenceType;
  recurrenceUntilUtc: string;
};

export type CalendarBooking = {
  bookingId: string;
  roomId: string;
  roomName: string;
  userId: string;
  userName: string;
  startTimeUtc: string;
  endTimeUtc: string;
  purpose: string;
  status: BookingStatus;
  recurringBookingSeriesId: string | null;
};
